### PR TITLE
Fix formatting + always default to 1 decimal place NC-694

### PR DESCRIPTION
**Pull request by olslash**
@drakonen @kaareal 

This fixes a bunch of d3.format specifiers that were presumably written for an old D3 version and weren't functioning properly. It also tries to make number formatting more consistent across the codebase (by using a new helper), and defaults decimals to 1 place by default.

Would appreciate help checking the edge cases -- i gave it a pretty good look but may have missed one or two.

### CODE REVIEW

File: src/visualization/helpers/format.js [hunk 1]
```diff
@@ -53,5 +53,32 @@ define(['../../core'], function(Openhose){
         return Openhose.Rickshaw.Fixtures.Number.formatKMBT(x);
       }
     },
+    numberFormatter: function(precision) {
+      precision || (precision = 1);
+
+      // first round, so if the number is whole, we don't get trailing zeroes
+      var rounded = Openhose._.partial(d3.round, _, precision);
```
Comment (drakonen) [inline src/visualization/helpers/format.js]: Second _ needs to be Openhose._ I think

File: serve/unit/visualization-helpers.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+define([
+  'core',
+  '../src/visualization/helpers/format'
+], function(Openhose, formatHelpers) {
+  describe('test format helpers', function() {
+    describe("test numberFormatter()", function() {
+      it("should return a function", function() {
+        assert(typeof formatHelpers.numberFormatter(2) === 'function');
```
Comment (drakonen) [inline serve/unit/visualization-helpers.js]: Do we have assert.typeOf() ?

File: serve/unit/visualization-helpers.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          [f(-2.1123, 2), '-2.11'],
+          [f(2.116, 2), '2.12'],
+          [f(2000000, 2), '2,000,000'],
+          [f(2000000.123, 2), '2,000,000.12'],
+          [f(2.123456, 5), '2.12346'],
+          [f(2.123456), '2.1'] // default to 1 place
+        ];
+
+        tests.forEach(function(test) {
+          assert(test[0] === test[1], 'expected ' + test[1] + ', got ' + test[0]);
```
Comment (drakonen) [inline serve/unit/visualization-helpers.js]: Better use assert.equal

**Comment by drakonen** [comment]
You are not consistent with the formatHelper required name.

Otherwise looks awesome! thanks!
:+1: :godmode:

**Comment by olslash** [comment]
thx

**Pull request merged**